### ISSUE TITLE
Integrated way of downgrading neos

### ISSUE THREAD

**Issue by 3x1t-5tyl3** [Neos-Metaverse/NeosPublic#2911]
### Is your feature request related to a problem? Please describe.

Currently there's no way other than thirdparty software (such as https://github.com/SteamRE/DepotDownloader) to download an old repo from steam. This can be potentially untrustworthy. (At-least for me)

### Relevant issues

- Didn't find any.

### Describe the solution you'd like

A way (trough the launcher, command line, ect) for me to able to downgrade neos.

This would allow findign bugs with are regression based easier and allows for a safe and clean way to downgrade neos instead of having to rely on third-party solutions.

How this is done doesn't matter as long as I am able to specify the specific version.

### Describe alternatives you've considered

- SteamDepotDownloaders; which as mentioned above might be sketchy to give my Steam credentials into.

### Additional context

This has been on my mind for a while; I'd love to do more regression based bug finding which can be hard when neos is always up-to-date and steam doesn't come with a good way to downgrade.

**Comment by shiftyscales** [comment]
There are plans for a dedicated Neos Launcher/standalone updater known as Neosine which will support a variety of features/functionalities.

This will also be useful for the sake of QC testing/isolating when a regression has occurred as you've noted, as well as running concurrent developmental branches so that major features don't need to hold up mainline development/bug fixes.

There has been considerable work done in the backgrounds towards automation of the build pipeline which would be a necessary step towards this goal.

The launcher will also be useful for things such as automatic updating/restarting of headless clients, and removes the dependency on Steam so that Neos can be distributed in commercial/professional settings. This is detailed on our roadmap. https://github.com/Neos-Metaverse/NeosPublic/projects/1#card-35406676

**Comment by shadowpanther** [comment]
> as well as running concurrent developmental branches
Hope it'd be done in such a way that several different builds could co-exist on one machine without the need to constantly up/down-grade the client. Optionally the ability to run several copies of Neos in parallel.

**Comment by shiftyscales** [comment]
> 
> 
> > as well as running concurrent developmental branches
> 
> Hope it'd be done in such a way that several different builds could co-exist on one machine without the need to constantly up/down-grade the client. Optionally the ability to run several copies of Neos in parallel.

Database management would definitely be helpful too.

**Comment by shadowpanther** [comment]
Yeah, database/cache/assets cleanup might also be a part of the launcher/updater, alongside with creation and management of different profiles (that would be required for running multiple copies in parallel). I already have an issue open on the database cleanup too - #2473.

**Comment by shiftyscales** [comment]
> 
> 
> Yeah, database/cache/assets cleanup might also be a part of the launcher/updater, alongside with creation and management of different profiles (that would be required for running multiple copies in parallel). I already have an issue open on the database cleanup too - #2473.

There is also #58 as well which is a generalized issue for configuring database/cache locations, as well as cache management.

These are not strictly necessary for the launcher, but would be nice to handle via the launcher when they can be implemented/prioritized.

**Comment by Frooxius** [comment]
This is planned and already partly in progress. It's quite a large task, because it requires us to roll our own update and distribution service on the Neos cloud, so you can manage all the necessary builds, but also so they can be automated - automatically spinning up and updating headless for example. You'll be able to have multiple versions installed at the same time too.

We'll probably be focusing on this one a lot more soon, since the manual tasks involved with this are eating a lot into actual development time, especially since we have more people on the team now than before.
